src/overlays/actors/ovl_En_Zf/z_en_zf.c

```diff
@@ -82,115 +151,2291 @@ static ColliderQuadInit D_80B4A204 = {
+        EnZf_FindNextPlatformAwayFromPlayer(&this->actor.world.pos, this->curPlatform, this->homePlatform, globalCtx);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnZf_SetupAction(this, EnZf_SheatheSword);
+}
+
+void EnZf_SheatheSword(EnZf* this, GlobalContext* globalCtx) {
+    s16 yaw = Actor_WorldYawTowardPoint(&this->actor, &sPlatformPositions[this->nextPlatform]) + 0x8000;
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, yaw, 1, 1000, 0);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actor.world.rot.y = yaw - 0x8000;
+        EnZf_SetupHopAway(this, globalCtx);
+        this->swordSheathed = true;
+    }
+}
+
+void EnZf_SetupHopAndTaunt(EnZf* this) {
+    this->hopAnimIndex = 0;
+    Animation_MorphToPlayOnce(&this->skelAnime, sHoppingAnims[0], -4.0f);
+    this->action = ENZF_ACTION_HOP_AND_TAUNT;
+    this->actor.speedXZ = 0.0f;
+    this->unk_40C = 0.0f;
+    this->unk_408 = 0.0f;
+    EnZf_SetupAction(this, EnZf_HopAndTaunt);
+}
+
+void EnZf_HopAndTaunt(EnZf* this, GlobalContext* globalCtx) {
+    f32 lastFrame;
+    f32 maxDist = 400.0f;
+
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, (this->actor.yawTowardsPlayer + 0x8000), 1, 4000, 0);
+
+    // Upstairs
+    if (this->actor.world.pos.y >= 420.0f) {
+        maxDist = 250.0f;
+    }
+
+    // If player gets too close, run away
+    if ((this->actor.xzDistToPlayer < maxDist) && (this->hopAnimIndex != 1)) {
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        EnZf_SetupSheatheSword(this, globalCtx);
+    } else {
+        if (this->hopAnimIndex != 1) {
+            EnZf_DodgeRangedWaiting(globalCtx, this);
+        }
+
+        if (SkelAnime_Update(&this->skelAnime)) {
+            this->hopAnimIndex++; // move on to next animation
+
+            // Loop back to beginning
+            if (this->hopAnimIndex >= ARRAY_COUNT(sHoppingAnims)) {
+                this->hopAnimIndex = 0;
+            }
+
+            if ((this->unk_408 != 0.0f) || (this->unk_40C != 0.0f)) {
+                this->hopAnimIndex = 1;
+            }
+
+            lastFrame = Animation_GetLastFrame(sHoppingAnims[this->hopAnimIndex]);
+
+            switch (this->hopAnimIndex) {
+                case 0:
+                    this->actor.velocity.y = 0.0f;
+                    this->actor.world.pos.y = this->actor.floorHeight;
+                    break;
+
+                case 1:
+                    this->actor.velocity.y = this->unk_40C + 10.0f;
+                    this->actor.speedXZ = this->unk_408;
+                    this->unk_408 = 0.0f;
+                    this->unk_40C = 0.0f;
+                    break;
+
+                case 2:
+                    this->actor.world.pos.y = this->actor.floorHeight;
+                    lastFrame = 3.0f;
+                    break;
+
+                default:
+                    break;
```

Comment:
is this needed for matching?

Reply:
I think it's better to always have a default anyway, just in case.